lib/SILGen/SILGenExpr.cpp

```diff
@@ -5358,29 +5363,27 @@ RValue RValueEmitter::visitUnevaluatedInstanceExpr(UnevaluatedInstanceExpr *E,
 }
 
 RValue RValueEmitter::visitTapExpr(TapExpr *E, SGFContext C) {
-  // This implementation is not very robust; if TapExpr were to ever become
-  // user-accessible (as some sort of "with" statement), it should probably
-  // permit a full pattern binding, saving the unused parts and "re-structuring"
-  // them to return the modified value.
-
-  auto Var = E->getVar();
   auto VarType = E->getType()->getCanonicalType();
 
   Scope outerScope(SGF, CleanupLocation(E));
 
   // Initialize the var with our SubExpr.
-  auto VarInit =
-    SGF.emitInitializationForVarDecl(Var, /*forceImmutable=*/false);
+  auto VarInit = SGF.emitTemporary(SILLocation(E),
+                                   SGF.getTypeLowering(VarType));
   SGF.emitExprInto(E->getSubExpr(), VarInit.get(), SILLocation(E));
 
+  auto VarAddress = VarInit->getManagedAddress();
+  SmallVector<std::unique_ptr<SILGenFunction::OpaqueValueRAII>, 5> opaqueValues;
+
+  for (auto use : E->getUses())
```

Comment:
At one point SILGen allowed an OVE to be referenced multiple times. I wonder if it would be difficult to bring that back. We'd still want them to be unique in most situations, but TapExpr could be an exception.

Reply:
A third "multiple use" bit, I suppose?